homeassistant/components/switcher_kis/quality_scale.yaml

```diff
@@ -0,0 +1,78 @@
+rules:
```

Comment:
Why do we have buttons for the swing and swing in the climate??

Reply:
Switcher Breeze is a generic device for controlling air conditioners, some of them has separate swing on/off commands with no feedback of the current state and some report the state of the swing. If there is a feedback we create a swing feature under the climate, if there is no feedback and two buttons are needed (one for on and one for off) we create button entities.

Climate:
https://github.com/home-assistant/core/blob/839eb0fe143a604cf2b8e97a7c0ee4336977e101/homeassistant/components/switcher_kis/climate.py#L113

Button:
https://github.com/home-assistant/core/blob/839eb0fe143a604cf2b8e97a7c0ee4336977e101/homeassistant/components/switcher_kis/button.py#L76

Reply:
I don't know how to implement this as assumed state, how would I know what is the current state of the swing? it can change outside of HA and in this case if it is not a button the user will not be able to change it.
For example:
Swing is ON, but HA doesn't know about it and show this as OFF, user want to enable Swing, the user would need to disable swing and enable it again.

Reply:
Justo to add more info, for switch it does work, because the frontend visualize this correctly, but swing feature is a selector and doesn't have an option for assumed state.

Reply:
I will check it just to answer here (I am pretty sure it doesn't work for Swing), but note that only the Swing has no state report for certain AC types, all other features of the Climate has state report, so setting Assumed state for the whole entity would not be correct.

Reply:
I have tested setting the `_attr_assumed_state` to `True` and didn't notice any change in the frontend, I did a step further and verified that the climate entity doesn't have any reference for assumed state:
https://github.com/home-assistant/frontend/blob/dev/src/data/climate.ts
Example from media player:
https://github.com/home-assistant/frontend/blob/6464c2b602ec44a30f1cdf5269a355f2519db89b/src/data/media-player.ts#L306

In any case the assumed state is only for the main entity state and not for sub features